### PR TITLE
Remember browse position when re-entering Seadroid via Launcher icon

### PR DESCRIPTION
**Pull request by forouher**
When tapping on the Seadroid Launcher icon, Seadroid currently always reverts back to
its "Home screen" (the list of repositories). Even if Seadroid was
already running and the user had already entered a specific folder.

This behaviour differs from most other apps. Usually the state of the app is preserved
if one re-enters it via the Launcher app.

This patch tries to do exactly that. The key place is BrowserActivity.onNewIntent()
In this method we have to decide whether to kill the old BrowserActivity and create a new
one; or to just keep the old one running.

Previously this was decided on the given account information. However this does not work
with the Launcher, as the account info is null if the intent comes from the Launcher.

I decided to use the Intent category/action for making that decision. If it is a MAIN
action, than it was called by the Launcher and the old instance shall be retained.

Therefore I had to convert UnlockGesturePasswordActivity and AccountsActivity to call
the BrowseActivity directly by its class name and not the action.

Note that a shorter fix would have been:

     -        if (!account.equals(selectedAccount)) {
     +        if (server == null || !account.equals(selectedAccount)) {

However BrowseActivity is started from a lot places, and I was a bit scared of creating new
bugs. This patch should hopefully have more controlled side effects.

### CODE REVIEW

**Comment by Logan676** [comment]
@forouher thanks for your awesome patch, a better solutions than #413.

**Comment by forouher** [comment]
Oh, another race condition. I should start reading open PRs before writing up patches. ;)

**Comment by shoeper** [comment]
Great! Just noticed this as "bug" as well.

File: src/com/seafile/seadroid2/ui/activity/AccountsActivity.java [hunk 1]
```diff
@@ -235,9 +235,8 @@ public void onBackPressed() {
         Account account = accountManager.getCurrentAccount();
         if (account == null) {
             // force exit when current account was deleted
-            Intent i = new Intent();
-            i.setAction(Intent.ACTION_MAIN);
-            i.addCategory(Intent.CATEGORY_HOME);
+            Intent i = new Intent(this, BrowserActivity.class);
+            i.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
```
Comment (Logan676) [inline src/com/seafile/seadroid2/ui/activity/AccountsActivity.java]: This fix is not correct, reproduce the error as following:
1. Open Seafile App for Android
2. Go to Settings -> Sign out -> click Confirm button, then it navigates to Accounts list
3. Choose and click one Account, re-enter password -> Login, then it navigates to the LIBRARY tab
4. Browsing file, enter a folder in order to make some browsing history
5. Click HOME button, then click the Launcher icon
~~6. Restore the browsing progress (correct)~~
6. It shows the Account list (incorrect)

Comment (forouher) [inline src/com/seafile/seadroid2/ui/activity/AccountsActivity.java]: Yes, you're right. But as you've already merged the alternative patch, I'm just closing this PR.

**Comment by forouher** [comment]
Closing this, as an alternative patch was already merged (please reopen if you're still interested in resurrecting this).

**Comment by Logan676** [comment]
@forouher still interested in this patch if you can resolve the commented issue, thanks.

**Comment by forouher** [comment]
okay, i'll look into it,

**Comment by forouher** [comment]
added some patches that should fix the reported issue, as well as some related issues that popped up during testing. I rebased the branch on 1.9.1.

**Comment by Logan676** [comment]
@forouher I have tested the patch and noticed that when logged out and relogined the account, the working flow goes to Account List, before it goes to Browsing page. This is not related to restoring browsing history as this PR proposed to I think.

**Comment by Logan676** [comment]
> better to return from the AccountDetailActivity back to the AccountActivity, and start the BrowserActivity from there.

I feel a little bit confused about that because not all of AccountDetailActivity were made to go back to the AccountActivity. e.g. account list --> edit account (switch account) --> start a new BrowsingActivity.

Below is what I have tested
- [x]  sign out current account -->  login again --> make some browsing history -->  home screen --> launcher icon --> browsing history restored
- [x] make some browsing history --> home screen --> launcher icon --> browsing history restored
- [x] setup gesture lock --> make some browsing history --> home screen --> wait for 5+ mins later for lock expired -->  screen locked --> unlock screen --> browsing history restored
- [ ] account list --> add new account --> back to account list (before it starts a new BrowsingActivity) and sorry this is **not allowed**.
- [x] account list --> edit account (switch account) --> start a new BrowsingActivity

**Comment by forouher** [comment]
* [ ] account list --> add new account --> back to account list (before it starts a new BrowsingActivity) and sorry this is not allowed.

I agree, that's not what I intended. I'm currently working on another patch to refactor the camera upload. Once that is done, I'll come back here and fix this PR up.

**Comment by forouher** [comment]
@Logan676
I amended a patch to address the final issue you found. Please let me know if there are any more open issues with this PR.

File: res/values-zh-rCN/strings.xml [hunk 2]
```diff
@@ -233,7 +233,7 @@
     <item quantity="other">%1$d秒后重试</item>
   </plurals>
   <plurals name="lockscreen_access_pattern_failure_left_try_times">
-    <item quantity="other">密码错误，还可以再输入<xliff:g id="times">%1$d</xliff:g>次</item>
+    <item quantity="other">密码错误，还可以再输入%1$d次</item>
```
Comment (Logan676) [inline res/values-zh-rCN/strings.xml]: Why has this patch introduced so many i18n modifications, please rebase the develop branch.

**Comment by forouher** [comment]
That was commit aa71bc8c9f51c8ff063a2287dbbb9d106bc8d5a8, that had somehow slipped into this PR. Sorry about that, I've rebased it on develop.

**Pull request merged**